### DISCUSSION TITLE
Micronaut 4.6.6 context propagation issue with project reactor

### DISCUSSION THREAD

**Discussion by SushmaReddyLoka** [micronaut-projects/micronaut-core#11346] [Q&A]
We upgraded to micronaut 4.6.6 and are seeing similar issue with project reactor calls, we have logs in doOnSuccess and context is missing in them. Also after switching back from reactor thread to default-nioEventLoopGroup-xx, the context which was originally set on default-nioEventLoopGroup-xx is also lost.

First we upgraded to 4.6.6 and observed that MDC.getCopyOfContextMap() was returning null altogether so we went ahead and added implementation("io.micrometer:context-propagation"), now we don't see NPE as MDC is having PropagationContext in it, but the rest of context is still lost.

We have been extending reactor.core.CoreSubscriber, and copying the currentContext() to MDC so far for context propogation.

**Comment by sdelamo** [comment]
@dstepanov any idea?

**Reply by dstepanov** [reply]
What is the rest of context?
